python/rapidsmpf/rapidsmpf/streaming/core/fanout.pyx

```diff
@@ -0,0 +1,77 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from libcpp.memory cimport make_unique
+from libcpp.utility cimport move
+from libcpp.vector cimport vector
+
+from rapidsmpf.streaming.core.channel cimport Channel
+from rapidsmpf.streaming.core.context cimport Context
+from rapidsmpf.streaming.core.fanout cimport FanoutPolicy
+from rapidsmpf.streaming.core.node cimport CppNode, cpp_Node
+
+
+def fanout(Context ctx, Channel ch_in, list chs_out, FanoutPolicy policy):
+    """
+    Broadcast messages from one input channel to multiple output channels.
+
+    The node continuously receives messages from the input channel and forwards
+    them to all output channels according to the selected fanout policy.
+
+    Each output channel receives a shallow copy of the same message; no payload
+    data is duplicated. All copies share the same underlying payload, ensuring
+    zero-copy broadcast semantics.
+
+    Parameters
+    ----------
+    ctx : Context
+        The node context to use.
+    ch_in : Channel
+        Input channel from which messages are received.
+    chs_out : list[Channel]
+        Output channels to which messages are broadcast.
+    policy : FanoutPolicy
+        The fanout strategy to use (see FanoutPolicy).
```

Comment:
This isn't really documented in python I think. Can we have a sentence here describing the two options?